plothist/comparison.py

```diff
@@ -0,0 +1,341 @@
+import numpy as np
+import scipy.stats as stats
+
+
+def _is_unweighted(hist):
+    """
+    Check whether a histogram is unweighted.
+
+    Parameters
+    ----------
+    hist : boost_histogram.Histogram
+        The histogram to check.
+
+    Returns
+    -------
+    bool
+        True if the histogram is unweighted, False otherwise.
+    """
+    return np.allclose(hist.variances(), hist.values())
+
+
+def _get_poisson_uncertainties(data_hist):
+    """
+    Get Poisson asymmetrical uncertainties for a histogram.
+
+    Parameters
+    ----------
+    data_hist : boost_histogram.Histogram
+        The histogram.
+
+    Returns
+    -------
+    uncertainties_low : numpy.ndarray
+        The lower uncertainties.
+    uncertainties_high : numpy.ndarray
+        The upper uncertainties.
+
+    Raise
+    -----
+    ValueError
+        If the histogram is weighted.
+
+    """
+    if not _is_unweighted(data_hist):
+        raise ValueError(
+            "Poisson uncertainties make sense only for unweighted histograms."
```

Comment:
After a lot of discussions, this sentence is not unanimous. We should either trust the user to use poisson correctly, or put documentation on why it is not possible.

Reply:
I am not saying it is impossible (I am not sure), but I do not know how to compute the Poisson uncertainties for an unweighted histogram. I changed the sentence.

Reply:
Since we don't know if it's possible, shouldn't we just raise a `Warning` except of a `ValueError`?